qiskit/primitives/__init__.py

```diff
@@ -17,59 +17,181 @@
 
 .. currentmodule:: qiskit.primitives
 
+The primitives are computational building blocks to be used in larger applications whose input 
+units, called primitive unified blocs (PUBs), require quantum resources to efficiently produce
+outputs for.
+
+Currently there are two types of primitives, whose abstractions, in their latest versions, are
+defined by :class:`~.BaseSamplerV2` and :class:`~.BaseEstimatorV2`. Samplers are responsible for
+accepting quantum circuits (or sweeps of values over parameterized circuits) and sampling from their
+classical output registers. Estimators accept combinations of circuits and observables (or sweeps
+thereof) to estimate expectation values of the observables.
+
+Qiskit implements a reference implementation for each of these abstractions, 
+:class:`~.StatevectorSampler` and :class:`~.StatevectorEstimator`.
+
 .. automodule:: qiskit.primitives.base.base_estimator
 .. automodule:: qiskit.primitives.base.base_sampler
 
-.. currentmodule:: qiskit.primitives
+Migration from Primitives V1 to V2
+==================================
 
-Estimator
-=========
+The formal distinction between the Primitives V1 and V2 APIs are the base classes from which 
+primitives implementations inherit, which are all listed at the bottom of the page. At a conceptual 
+level, however, here are some notable differences keep in mind when migrating from V1 to V2:
+
+1. The V2 primitives favour vectorized inputs, where single circuits can be grouped with 
+   vector-valued (or more generally, array-valued) specifications. Each group is called a 
+   primitive unified bloc (pub), and each pub gets its own result. For example, in the estimator, 
+   you can compare the following differences:
+
+   .. code-block:: python
+
+      # Favoured V2 pattern. There is only one pub here, but there could be more.
+      job = estimator_v2.run([(circuit, [obs1, obs2, obs3, obs4])])
+      evs = job.result()[0].data.evs
+
+      # V1 equivalent, where the same circuit must be provided four times.
+      job = estimator_v1.run([circuit, circuit, circuit, circuit], [obs1, obs2, obs3, obs4])
```

Comment:
```suggestion
      job = estimator_v1.run([circuit] * 4, [obs1, obs2, obs3, obs4])
```

This might be my preference. You can ignore.